html/includes/forms/schedule-maintenance.inc.php

```diff
@@ -129,6 +129,9 @@
             foreach ($_POST['maps'] as $target) {
                 $target = target_to_id($target);
                 $item   = dbInsert(array('schedule_id' => $schedule_id, 'target' => $target), 'alert_schedule_items');
+                if ($notes) {
+                    dbQuery('UPDATE `devices` SET `notes` = CONCAT(IFNULL(`notes`, ""), "' . date("Y-m-d H:i") . ' Alerts delayed:' .  $notes . '") WHERE `device_id` = ' . $target . ';');
```

Comment:
Don't use dbQuery like this, it is vulnerable to sql injection.  You should always use the parameters array.

Why not just fetch notes and then set it?

Will the notes ever be removed?

Reply:
MySQL can concatenate strings, so why don't use it ?

I changed to use dbupdate instead of dbquery, with parameters. So, needed to fetch notes before update.

Nop, never removed